src/decorators/transaction-id.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * A parameter containing the current transaction-id provided
+ * via HTTP request header "X-Transaction-Id".
+ */
+
+import { Reflector } from '@loopback/context';
+
+export function transactionIdHeader(resp: any) {
+	return function(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
+		// Get header here ... 
+		const header = 'Hello world header';
+		console.log('target =>', target);
+		console.log('propertyKey =>', propertyKey);
+		console.log('descriptor =>', descriptor);
+		return {
+			value: function(...args:any[]) {
+				const req: any = Reflector.getMetadata('http.request', )
```

Comment:
I tried writing a simple transactionIdHeader but ran into the issue of not being able to access the Context in a decorator to get the value for the header from `http.request`. Any ideas how this can be achieved? I'm guessing the right way for request associated transactions would be to do this through sequence?

Reply:
Here is the decorator implementation I had in mind:

```ts
export function transactionIdHeader() {
  return param.header.string('transactionId');
}
```

It delegates the implementation complexity to the built-in `@param` decorator.

Reply:
To answer your original question: the decorators cannot access any context, they are invoked at the time when the source code files are required (loaded by Node.js runtime), there is no context instance created at that time yet.

You can look at [`@param()`](https://github.com/strongloop/loopback-next/blob/210d8a13a60991c391433b9f9ea1885cca135ee0/packages/core/src/router/metadata.ts#L278-L317) implementation to understand how the `param` decorator stores metadata, and [`parseParams()`](https://github.com/strongloop/loopback-next/blob/210d8a13a60991c391433b9f9ea1885cca135ee0/packages/core/src/parser.ts#L56-L64) sequence actions to understand how this metadata is used.